tensorflow/python/data/experimental/ops/readers.py

```diff
@@ -209,9 +209,10 @@ def _maybe_shuffle_and_repeat(
   """Optionally shuffle and repeat dataset, as requested."""
   if num_epochs != 1 and shuffle:
     # Use shuffle_and_repeat for perf
-    return dataset.apply(
-        shuffle_ops.shuffle_and_repeat(shuffle_buffer_size, num_epochs,
-                                       shuffle_seed))
+    def _apply_fn(input_dataset):  # pylint: disable=missing-docstring
```

Comment:
you can just return `shuffle_ops._ShuffleAndRepeatDataset(...)` instead of doing the `apply`

Reply:
and for consistency, it might make sense to do the same for `parallel_interleave` (instead of the `_parallel_interleave` indirection you introduced)